sol/contracts/ics23Proof.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.2;
+pragma experimental ABIEncoderV2;
```

Comment:
Pragma `ABIEncoderV2` is not required since 0.8 release, I think we can remove it

> You can choose to use the old behaviour using pragma abicoder v1;. The pragma pragma experimental ABIEncoderV2; is still valid, but it is deprecated and has no effect. If you want to be explicit, please use pragma abicoder v2; instead.

Source ^^: https://docs.soliditylang.org/en/v0.8.0/080-breaking-changes.html